pkg/graph/parser/parser.go

```diff
@@ -236,7 +236,7 @@ func parseScalarTypes(field interface{}, _ *spec.Schema, path string, expectedTy
 	// perform type checks for scalar types
 	switch {
 	case slices.Contains(expectedTypes, "number"):
-		if _, ok := field.(float64); !ok {
+		if !isNumber(field) {
```

Comment:
So if i understand correctly, in openapispecv3, `number` values can be either `int` or `float` in go?

Reply:
Actually, Open api v3 number refers to float and double. https://spec.openapis.org/oas/v3.0.0.html#openapi-specification:~:text=The%20formats%20defined%20by%20the%20OAS%20are%3A
Which might make the issue not only related to this method only, because the error i got refers that it expected to get the type 'number' but received an int64: Defining a resource with below resource definition fails on: "expected number type for path renotifyInterval, got int64".
parseScalarTypes func gets 'number' at expectedTypes slice. But the 'field' variable held an int64 type.

Reply:
The open api resource field type is really a number: kubectl get crd tests.synthetics.datadog.upbound.io -o yaml
`
renotifyInterval:
  description: (Number) Specify a renotification frequency
    in minutes. Values available by default are 0, 10,
    20, 30, 40, 50, 60, 90, 120, 180, 240, 300, 360,
    720, 1440. Defaults to 0. Specify a renotification
    frequency in minutes. Values available by default
    are `0`, `10`, `20`, `30`, `40`, `50`, `60`, `90`,
    `120`, `180`, `240`, `300`, `360`, `720`, `1440`.
    Defaults to `0`.
  type: number
`

Which leaves the question, why `resource interface{}` type is int64.

Reply:
@a-hilaly did you get the chance to check this?

Reply:
Hey @khalilj ! Thanks for the making the changes. Yep LGTM, i'm just running some tests with float/int to understand more why `resource interface{}` type is int64 - otherwise this is ready to go :)

Reply:
Thanks @a-hilaly

Reply:
Hi @a-hilaly , did you get the chance to test it?

Reply:
After a quick deep dive, looks like this is related to the JSON unmarshallers client-go is using https://github.com/kubernetes/apimachinery/blob/master/pkg/runtime/serializer/json/json.go - i think this is ready to go. Thanks a lot @khalilj !

Reply:
Great news @a-hilaly , Thanks!